strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/pypi/PypiArtifactControllerTest.java

```diff
@@ -0,0 +1,272 @@
+package org.carlspring.strongbox.controllers.layout.pypi;
+
+import org.carlspring.strongbox.config.IntegrationTest;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.rest.common.PypiRestAssuredBaseTest;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.testing.artifact.ArtifactManagementTestExecutionListener;
+import org.carlspring.strongbox.testing.artifact.PypiTestArtifact;
+import org.carlspring.strongbox.testing.repository.PypiTestRepository;
+import org.carlspring.strongbox.testing.storage.repository.RepositoryManagementTestExecutionListener;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+
+import io.restassured.http.ContentType;
+
+/**
+ * 
+ * @author ankit.tomar
+ *
+ */
+@IntegrationTest
+public class PypiArtifactControllerTest extends PypiRestAssuredBaseTest
+{
+
+    @Override
+    @BeforeEach
+    public void init()
+        throws Exception
+    {
+        super.init();
+    }
+
+    @Test
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    public void testUploadPackageFlow(@PypiTestRepository(repositoryId = REPOSITORY_RELEASES, storageId = REPOSITORY_STORAGE) Repository repository,
+                                      @PypiTestArtifact(repositoryId = REPOSITORY_RELEASES, storageId = REPOSITORY_STORAGE, id = "hello_world_pypi", versions = "1.3") Path packagePath)
```

Comment:
@anki2189 you shouldn't use `repositoryId`  and `storageId` for `@PypiTestArtifact` in case of upload test.

Reply:
Noted.